pages/products/index.js

```diff
@@ -59,49 +57,70 @@ const StyledButtonPosition = styled.div`
 export default function Products() {
-  const [products, setProducts] = useAtom(initialProducts);
   const router = useRouter();
 
+  const { data: storedProducts, mutate: changeProducts } =
+    useSWR(`/api/products`);
+
+  async function handleDeleteProduct(id) {
+    try {
+      await fetch(`/api/products/${id}`, {
+        method: "DELETE",
+      });
+      changeProducts();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
   return (
     <>
       <StyledAllCards>
-        {products.map((product) => {
-          return (
-            <StyledProductCard key={product.code}>
-              <StyledProductName>{product.name}</StyledProductName>
-              <StyledButtonPosition>
-                <StyledLink href={`/product/${product.code}`}>
-                  <SVGIcon variant="details" width="26px" />
-                </StyledLink>
-                <StyledDeleteButton
-                  variant="delete"
-                  width="30px"
-                  type="button"
-                  onClick={() => {
-                    if (products.length === 1) {
-                      setProducts(RESET);
-                    } else {
-                      const newProducts = products.filter(
-                        (item) => item.name !== product.name
-                      );
-                      setProducts(newProducts);
-                    }
-                  }}
-                >
-                  <SVGIcon variant="delete" width="26px" />
-                </StyledDeleteButton>
-              </StyledButtonPosition>
-              <div>
-                <StyledImage
-                  width={1000}
-                  height={1000}
-                  src={product.url}
-                  alt={product.name}
-                />
-              </div>
-            </StyledProductCard>
-          );
-        })}
+        {storedProducts &&
+          storedProducts.map((product) => {
+            return (
+              <StyledProductCard key={product.code}>
+                <StyledProductName length={product.name.length}>
+                  {product.name}
+                </StyledProductName>
+                <StyledButtonPosition>
+                  <StyledLink
+                    aria-label="link to product details"
+                    href={`/product/${product.code}`}
+                  >
+                    <SVGIcon
+                      aria-label="link to product details"
+                      variant="details"
+                      width="26px"
+                    />
+                  </StyledLink>
+                  <StyledDeleteButton
+                    aria-label="delete product"
+                    variant="delete"
+                    width="30px"
+                    type="button"
+                    onClick={() => {
+                      handleDeleteProduct(product._id);
+                    }}
+                  >
+                    <SVGIcon
+                      aria-label="delete product"
```

Comment:
Do you need an aria-label for your icon if the button already has one?

Reply:
maybe not :)